src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -787,33 +790,48 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, in Val
                 return true;
             }
 
-            using INodeStorage.WriteBatch writeBatch = _nodeStorage.StartWriteBatch();
+            ActionBlock<INodeStorage.WriteBatch> actionBlock =
+                new ActionBlock<INodeStorage.WriteBatch>((batch) => batch.Dispose());
 
-            void DoAct(KeyValuePair<HashAndTinyPath, Hash256> keyValuePair)
+            INodeStorage.WriteBatch writeBatch = _nodeStorage.StartWriteBatch();
+            try
             {
-                HashAndTinyPath key = keyValuePair.Key;
-                if (_pastPathHash.TryGet(key, out ValueHash256 prevHash))
+                int round = 0;
+                foreach (KeyValuePair<HashAndTinyPath, Hash256> keyValuePair in persistedHashes)
                 {
-                    TreePath fullPath = key.path.ToTreePath(); // Micro op to reduce double convert
-                    Hash256? hash = key.addr == default ? null : key.addr.ToCommitment();
-                    if (CanRemove(hash, key.path, fullPath, prevHash, keyValuePair.Value))
+                    HashAndTinyPath key = keyValuePair.Key;
+                    if (_pastPathHash.TryGet(key, out ValueHash256 prevHash))
                     {
-                        Metrics.RemovedNodeCount++;
-                        writeBatch.Remove(hash, fullPath, prevHash);
+                        TreePath fullPath = key.path.ToTreePath(); // Micro op to reduce double convert
+                        if (CanRemove(key.addr, key.path, fullPath, prevHash, keyValuePair.Value))
+                        {
+                            Metrics.RemovedNodeCount++;
+                            writeBatch.Set(key.addr, fullPath, prevHash, default, WriteFlags.DisableWAL);
+                            round++;
+                        }
+                    }
+
+                    // Batches of 1000
+                    if (round > 1000)
```

Comment:
did you experiment with other batch sizes?

Reply:
Testing with 256 atm